### PR TITLE
Fixed exp stages ignoring last stage

### PR DESCRIPTION
**Pull request by EPuncker**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Fixed exp stages ignoring last stage

### Context

> if you have x5 in config
> (not staged rate)
> last stage will be ignored and use that value instead
> because it checks min < level < max
> and max is 0
> in last stage
> so it never satisfies
> and defaults to RATE_EXP

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by ranisalt** [comment]
I would fix it differently, by setting the default to maximum possible value when loading:

```cpp
			if (auto maxLevelAttribute = stageNode.attribute("maxlevel")) {);
				maxLevel = pugi::cast<uint32_t>(maxLevelAttribute.value());
			} else {
				maxLevel = std::numeric_limits<uint32_t>::max();
			}
```

To avoid branching

**Comment by Zbizu** [comment]
speaking of exp stages,
these three variables probably aren't used by anything anymore:
https://github.com/otland/forgottenserver/blob/e288183a5a459f5c5f7a470dcd9001e6cd745de0/src/game.h#L586

**Comment by ranisalt** [comment]
@EPuncker the code goes in loading the stages, not where you put it

File: src/configmanager.cpp [hunk 1]
```diff
@@ -328,7 +328,8 @@ bool ConfigManager::getBoolean(boolean_config_t what) const
 float ConfigManager::getExperienceStage(uint32_t level) const
 {
 	auto it = std::find_if(expStages.begin(), expStages.end(), [level](ExperienceStages::value_type stage) {
```
Comment (ranisalt) [inline src/configmanager.cpp:330]: While at that you can change it here to use `auto`
```suggestion
	auto it = std::find_if(expStages.begin(), expStages.end(), [level](auto stage) {
```

And I THINK this is possible:
```suggestion
	auto it = std::find_if(expStages.begin(), expStages.end(), [level](auto&& [minLevel, maxLevel]) {
```
or
```suggestion
	auto it = std::find_if(expStages.begin(), expStages.end(), [level](auto stage) {
            auto&& [minLevel, maxLevel] = stage;
```

Comment (EPuncker) [reply src/configmanager.cpp:330]: feel free to commit on this branch (please)

**Review by ranisalt — APPROVED** [review]
image

**Comment by nekiro** [comment]
I think it still doesnt work. With last stages without maxLevel it returned 0 for me, which failed to find the stage and defaulted back to flat expRate.
I added check for 0 and it works, but most likely that getField doesnt actually do what it should do with that defaultValue

**Pull request merged**